Upstream bugId: #3277
Synopsis: Info panel can't obtain file system statistics
Bug record: https://www.midnight-commander.org/ticket/3277

--- mc-4.8.14/src/filemanager/mountlist.c.~1~	2015-03-20 21:06:04.000000000 +0300
+++ mc-4.8.14/src/filemanager/mountlist.c	2015-08-19 21:00:45.270143719 +0300
@@ -1051,9 +1051,9 @@
         char *table = MNTTAB;
         FILE *fp;
         int ret;
-        int lockfd = -1;
 
 #if defined F_RDLCK && defined F_SETLKW
+        int lockfd = -1;
         /* MNTTAB_LOCK is a macro name of our own invention; it's not present in
            e.g. Solaris 2.6.  If the SVR4 folks ever define a macro
            for this file name, we should use their macro name instead.
@@ -1108,8 +1108,10 @@
             ret = fclose (fp) == EOF ? errno : 0 < ret ? 0 : -1;
         }
 
+#if defined F_RDLCK && defined F_SETLKW
         if (lockfd >= 0 && close (lockfd) != 0)
             ret = errno;
+#endif
 
         if (ret >= 0)
         {
